discussment-examples/spring/src/main/java/org/danekja/discussment/spring/CoreConfiguration.java

```diff
@@ -0,0 +1,110 @@
+package org.danekja.discussment.spring;
+
+import org.danekja.discussment.core.accesscontrol.dao.PermissionDao;
+import org.danekja.discussment.core.accesscontrol.dao.jpa.PermissionDaoJPA;
+import org.danekja.discussment.core.accesscontrol.service.AccessControlManagerService;
+import org.danekja.discussment.core.accesscontrol.service.AccessControlService;
+import org.danekja.discussment.core.accesscontrol.service.PermissionManagementService;
+import org.danekja.discussment.core.accesscontrol.service.impl.PermissionService;
+import org.danekja.discussment.core.dao.*;
+import org.danekja.discussment.core.dao.jpa.*;
+import org.danekja.discussment.core.service.*;
+import org.danekja.discussment.core.service.imp.*;
+import org.danekja.discussment.spring.core.dao.UserDao;
+import org.danekja.discussment.spring.core.dao.jpa.UserDaoJPA;
+import org.danekja.discussment.spring.core.service.UserService;
+import org.danekja.discussment.spring.core.service.imp.DefaultUserService;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+
+@Configuration
+@EnableTransactionManagement
+public class CoreConfiguration {
+
+    @PersistenceContext
+    private EntityManager em;
```

Comment:
Make GenericDaoJPA provide two constructors - one taking the EntityManager as parameter (current situation). Add one constructor, which takes only the class and no em. Add `@PersistenceContext` annotated EntityManager attribute to the GenericDaoJPA.

That way - if user wants to use own entity manager, he can (current constructor), if he wants to use container-manged one, he can - he simply creates his dao without any arguments (e.g. `new CategoriyDaoJPA()`).

--> remove the entity manager attribute from this configuration as it is no longer necessary.